### PR TITLE
chore(monocdk): support barrel imports

### PR DESCRIPTION
**Pull request by eladb**
## Commit Message

chore(monocdk): support barrel imports (#6996)

This change enables an alternative syntax for importing submodules from monocdk:

    import * as s3 from 'monocdk-experiment/aws-s3';

In addition to the currently supported syntax:

    import { aws_s3 as s3 } from 'monocdk-experiment';

The new syntax will make it easy to migrate from existing CDK codebase and also makes it easier to namespace submodule imports by consumers.

To that end, we hoisted the submodules from `src` to be direct subdirectories of the package (otherwise, the barrel import would have been `monocdk-experiment/src/aws-s3`).

Credits: @CaerusKaru

## End of Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@monocdk-experiment/assert/package.json [hunk 1]
```diff
@@ -46,7 +47,7 @@
     "pkglint": "0.0.0",
     "ts-jest": "^25.2.0",
     "@monocdk-experiment/rewrite-imports": "0.0.0",
-    "monocdk-experiment": "0.0.0",
+    "monocdk-experiment": "file:../../monocdk-experiment",
```
Comment (RomainMuller) [inline packages/@monocdk-experiment/assert/package.json]: I don't understand why this change is there?

File: packages/monocdk-experiment/build.sh [hunk 2]
```diff
@@ -34,5 +34,5 @@ mkdir -p dist/js
 cp ${tarball} dist/js
 
 # copying src/ so this module will also work as a local dependency (e.g. for modules under @monocdk-experiment/*).
-rm -fr src
-rsync -av ${unpacked}/src/ src/
+rm -fr staging
+rsync -av ${unpacked}/ staging/
```
Comment (RomainMuller) [inline packages/monocdk-experiment/build.sh]: Might be faster / avoids wearing SSDs over too much (but could literally also not work, I haven't tried):

```suggestion
# copying staging/ so this module will also work as a local dependency (e.g. for modules under @monocdk-experiment/*
rsync -av --delete ${unpacked}/ staging/
```

File: packages/monocdk-experiment/deps.js [hunk 3]
```diff
@@ -14,6 +14,10 @@ const modules = fs.readdirSync(root);
 let errors = false;
 
 for (const dir of modules) {
+  if (dir.startsWith('.')) {
+    continue;
+  }
+
```
Comment (RomainMuller) [inline packages/monocdk-experiment/deps.js]: Why is it necessary and wasn't before?

File: packages/monocdk-experiment/gen.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ const exclude_files = [
 async function main() {
   const outdir = path.join(await fs.mkdtemp(path.join(os.tmpdir(), 'monocdk-')), 'package');
 
-  const srcdir = path.join(outdir, 'src');
+  const srcdir = outdir;
```
Comment (RomainMuller) [inline packages/monocdk-experiment/gen.js]: Why not just use `outdoor` instead of aliasing?

File: packages/monocdk-experiment/build.sh (hunk 2 above)
Comment (eladb) [reply packages/monocdk-experiment/build.sh]: Now i am realizing I can just move the staging directory from tmp

File: packages/@monocdk-experiment/assert/package.json (hunk 1 above)
Comment (eladb) [reply packages/@monocdk-experiment/assert/package.json]: Will revert. Good point

File: packages/monocdk-experiment/gen.js (hunk 4 above)
Comment (eladb) [reply packages/monocdk-experiment/gen.js]: reverted

File: packages/monocdk-experiment/deps.js (hunk 3 above)
Comment (eladb) [reply packages/monocdk-experiment/deps.js]: reverted
